lib/os/json.c

```diff
@@ -549,6 +540,8 @@ static int arr_parse(struct json_obj *obj,
 
 	if (val) {
 		elements = (size_t *)((char *)val + elem_descr->offset);
+	} else {
+		field = (char *)field + elem_descr->offset;
```

Comment:
When the `field` pointer is updated and skipped over a (unused?) field, shouldn't the `last_elem` variable also be updated?

Reply:
In this new design `get_elem_size()` returns the actual size of the underlying object, so `last_elem` correctly reflects the address of the last element.

However I'm still not convinced the way `field` is adjusted when parsing the nested array (when parsing an array of array).

Reply:
> However I'm still not convinced the way `field` is adjusted when parsing the nested array (when parsing an array of array).

Indeed. I hope that'll be fixed soon by #50816.